src/libs/actions/IOU.js

```diff
@@ -111,8 +111,38 @@ function createIOUSplit(params) {
         });
 }
 
+/**
+ * Pays an IOU Report and then retrieves the iou and chat reports to trigger updates to the UI.
+ */
+function payIOUReport({
+    chatReportID, reportID, paymentMethodType,
+}) {
+    Onyx.merge(ONYXKEYS.IOU, {loading: true, error: false});
+    API.PayIOU({
+        reportID,
+        paymentMethodType,
+    })
+        .then((response) => {
+            if (response.jsonCode !== 200) {
+                throw new Error(response.message);
+            }
+            fetchChatReportsByIDs([chatReportID]);
+
+            // An iouReport that is being paid must be open, and open iouReports are always linked to the associated
+            // chatReport via its Onyx data. As this link must be kept updated, after fetching the report we must also
+            // update the chatReport link - see function for more info.
```

Comment:
The idea here was that the bulk of the explanation is tied to the function itself, rather than being repeated in each place that it was used. with a few modifications.

But I also think this was the weakest of the updated comments, so I've added a bit more details:

```
// If an iouReport is open (has an IOU, but is not yet paid) then we maintain a link between it and the
// associated chatReport in Onyx, simplifying IOU data retrieval and reducing necessary API calls when
// displaying IOU components. If the link isn't updated in this case, the paid IOU would still be shown
// to users as unpaid. The iouReport being fetched here must be open, because only an open iouReoport can
// be paid. Therefore, we should also update the chatReport link after fetching the iouReport.
```

Reply:
Maybe something like this...

>If an iouReport is open (has an IOU, but is not yet paid) then we must update the iouReportID in the chatReport object. If we do not do this the UI will incorrectly show our IOU as "unpaid". Therefore, we must fetch both iouReport and chatReport after the IOU has been paid.